src-electron/zcl/zcl-loader.js

```diff
@@ -195,7 +195,18 @@ async function qualifyZclFile(
       env.logDebug(
         `CRC missmatch for file ${pkg.path}, (${pkg.crc} vs ${actualCrc}) package id ${pkg.id}, parsing.`
       )
-      await queryPackage.updatePathCrc(db, filePath, actualCrc, parentPackageId)
+      // Reloading the zcl xml file when there is a CRC mismatch(changes) on it.
+      // This check makes sure that when a custom xml is edited, deleted and
+      // re-added from ZCL Extensions in ZAP UI then the custom xml package is
+      // updated in the backend to reflect the right configuration.
+      await queryPackage.deletePackagesByPackageId(db, pkg.id)
```

Comment:
This deleting of packages still sounds dangerous. How do you know there isn't another session using this package?

Reply:
This is done when the zcl loading happens. Is it possible to have loading happen when an existing session is still in progress?

Reply:
Sorry ignore me. This is the custom xml through zcl extension. I see your point.